cmd/node_agent/na/aws_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package na
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/awstesting/unit"
+)
+
+func initTestServer(path string, resp string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.RequestURI != path {
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+
+		_, _ = w.Write([]byte(resp))
+	}))
+}
+
+func TestIsProperPlatform(t *testing.T) {
+	server := initTestServer(
+		"/latest/meta-data/instance-id",
+		"instance-id",
+	)
+
+	c := ec2metadata.New(unit.Session, &aws.Config{Endpoint: aws.String(server.URL + "/latest")})
+	na := &awsPlatformImpl{client: c}
+	if !na.IsProperPlatform() {
+		t.Errorf("On Proper Platform: expected true")
+	}
+
+	server.Close()
+	if na.IsProperPlatform() {
+		t.Errorf("On Proper Platform: expected false")
+	}
+}
+
+func TestGetInstanceIdentityDocument(t *testing.T) {
+	server := initTestServer(
+		"/latest/dynamic/instance-identity/pkcs7",
```

Comment:
If possible, can you move this to a file in testdata/?

Reply:
will do

Reply:
I'll change it in next PR